src/ucp/proto/proto_multi.inl

```diff
@@ -62,12 +62,19 @@ ucp_proto_multi_max_payload(ucp_request_t *req,
                             const ucp_proto_multi_lane_priv_t *lpriv,
                             size_t hdr_size)
 {
-    size_t length   = req->send.state.dt_iter.length;
-    size_t max_frag = lpriv->max_frag - hdr_size;
+    size_t length    = req->send.state.dt_iter.length;
+    ssize_t max_frag = lpriv->max_frag - hdr_size;
     size_t max_payload;
 
-    ucs_assertv(lpriv->max_frag > hdr_size, "max_frag=%zu hdr_size=%zu",
-                lpriv->max_frag, hdr_size);
+    /* If header is greater than max_frag - that's ok, we allow it, but we send
+     * only the header data and 1 byte of payload.
+     * 1 byte is needed to distinguish between first and middle fragments that
+     * we do by comparing req->send.state.dt_iter.offset with 0.
+     * @see ucp_worker_set_max_am_header
+     */
```

Comment:
since we decided that no need to backport this fix to v1.18.x, how hard it will be to fix according to
```
         * TODO: fix generic AM based multi-fragment protocols, so that this
         * trick is not needed.
```

Reply:
done